emcc.py

```diff
@@ -1298,17 +1305,20 @@ def consume_arg_file():
           settings.GENERATE_DWARF = 1
         elif requested_level == 'source-map':
           settings.GENERATE_SOURCE_MAP = 1
+          settings.EMIT_NAME_SECTION = 1
           newargs[i] = '-g'
         else:
           # Other non-integer levels (e.g. -gline-tables-only or -gdwarf-5) are
           # usually clang flags that emit DWARF. So we pass them through to
           # clang and make the emscripten code treat it like any other DWARF.
           settings.GENERATE_DWARF = 1
+          settings.EMIT_NAME_SECTION = 1
         # In all cases set the emscripten debug level to 3 so that we do not
         # strip during link (during compile, this does not make a difference).
         settings.DEBUG_LEVEL = 3
     elif check_flag('-profiling') or check_flag('--profiling'):
       settings.DEBUG_LEVEL = max(settings.DEBUG_LEVEL, 2)
+      settings.EMIT_NAME_SECTION = 1
```

Comment:
Why all these `EMIT_NAME_SECTION` changes?  Can the be separated?

Reply:
It was because Binaryen was being used to strip the name section when it was run. But now we aren't running Binaryen by default, so the plumbing for this had to be fixed more properly. I can see if it can be separated easily.

Reply:
Yes, lets separate that out.  Maybe we can temporarily add `also_with_bigint` to some debug info tests to verify it?

Reply:
https://github.com/emscripten-core/emscripten/pull/23182